online-shop/src/app/services/cart-service/ShoppingCartService.service.ts

```diff
@@ -0,0 +1,94 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { OrderRequest } from 'src/app/shared/models/OrderRequest.model';
+import { Product } from 'src/app/shared/models/Product.model';
+import { ShoppingCartItem } from 'src/app/shared/models/ShoppingCartItem.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ShoppingCartService {
+  items: ShoppingCartItem[] = [];
+  private ordersUrl: string = 'http://localhost:3000/orders';
+  constructor(private httpClient: HttpClient) { }
+
+  addToCart(product: Product) {
+    if (this.checkIfInCartById(product.id) === false) {
```

Comment:
it's not mandatory to explicitly compare with false. Condition can be simplified (easy to read):
if (this.checkIfInCartById(product.id))
if condition will evaluate to true if value is not: null/undefined/empty_string/0/false